src/bind.c

```diff
@@ -271,8 +271,9 @@ static SEXP vec_cbind(SEXP xs, SEXP ptype, SEXP size, struct name_repair_opts* n
   R_len_t n = Rf_length(xs);
 
   // Find the common container type of inputs
-  SEXP containers = PROTECT(map(xs, &cbind_container_type));
-  ptype = PROTECT(cbind_container_type(ptype));
+  SEXP rownames = R_NilValue;
+  SEXP containers = PROTECT(map_with_data(xs, &cbind_container_type, &rownames));
+  ptype = PROTECT(cbind_container_type(ptype, &rownames));
```

Comment:
I guess making `map_with_data()` take `void* data` makes it generic, but seeing `cbind_container_type()` take `void* data` was pretty confusing at first. I really expected that to be a `SEXP` and was searching through `cbind_container_type()` to figure out why it was a `void*`.

Reply:
In general taking a `void*` signals the function is used as callback with a generic higher order function. I'm not sure making the `void*` function a wrapper around a `SEXP*` function would make things clearer overall because of the added noise of having multiple functions.